c10/xpu/test/impl/XPUTest.cpp

```diff
@@ -0,0 +1,69 @@
+#include <gtest/gtest.h>
+
+#include <c10/xpu/XPUFunctions.h>
+
+#define ASSERT_EQ_XPU(X, Y) \
+  {                         \
+    bool _isEQ = X == Y;    \
+    ASSERT_TRUE(_isEQ);     \
+  }
+
+bool has_xpu() {
+  return c10::xpu::device_count() > 0;
+}
+
+TEST(XPUTest, DeviceCount) {
+  if (!has_xpu()) {
+    return;
+  }
+
+#ifndef _WIN32
```

Comment:
If we won't support windows by current plan, we should remove it. If building happens on windows, we can raise error in cmake system.

Reply:
If build on windows, line 20-22 could not be included. why it could raise an error in cmake system?